pyproject.toml

```diff
@@ -32,10 +32,21 @@ dependencies = [
 [tool.hatch.version]
 path = "src/kagglehub/__init__.py"
 
+[project.optional-dependencies]
+dev = [
```

Comment:
should this be added to `tool.hatch.envs.default` instead?

Reply:
I was doing local development on my mac as follows:

```
pip -m venv .env
./.env/bin/activate
pip install -e .[dev]
```

After, I can run all the commands locally: `ruff fmt`, `ruff lint`

Reply:
You should not have to create and manage virtual environment manually. `hatch` is handling all of that.

Reply:
And then, you can run the convenience script I created for fmt / lint etc: https://github.com/Kaggle/kagglehub?tab=readme-ov-file#lint--format

Reply:
The fmt script for instance calls `black`, `ruff` and `style`: https://github.com/Kaggle/kagglehub/blob/9c25bbfe2bcfcce847cbb5ddbe080b607f63d153/pyproject.toml#L70-L74